engine/src/test/java/io/camunda/zeebe/engine/processing/incident/MultiInstanceIncidentTest.java

```diff
@@ -257,4 +260,84 @@ public void shouldUseTheSameLoopVariablesWhenIncidentResolved() {
             tuple(VariableIntent.UPDATED, "item", "3"),
             tuple(VariableIntent.UPDATED, "loopCounter", "3"));
   }
+
+  @Test
+  public void shouldResolveIncidentDueToInputCollection() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess("multi-task")
+                .startEvent()
+                .serviceTask(
+                    ELEMENT_ID,
+                    t ->
+                        t.zeebeJobType(jobType)
+                            .multiInstance(
+                                b ->
+                                    b.sequential()
+                                        .zeebeInputCollectionExpression(INPUT_COLLECTION)
+                                        .zeebeInputElement(INPUT_ELEMENT)))
+                .endEvent()
+                .done())
+        .deploy();
+
+    final var processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId("multi-task")
+            .withVariable(INPUT_COLLECTION, List.of(1, 2, 3))
+            .create();
+
+    final var activatedTask =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId(ELEMENT_ID)
+            .getFirst();
+
+    ENGINE
+        .variables()
+        .ofScope(activatedTask.getKey())
+        .withDocument(Collections.singletonMap(INPUT_COLLECTION, "not a list"))
+        .update();
+
+    completeNthJob(processInstanceKey, 1);
+
+    final var incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    // when
+    ENGINE
+        .variables()
+        .ofScope(activatedTask.getKey())
+        .withDocument(Collections.singletonMap(INPUT_COLLECTION, List.of(1, 2, 3)))
+        .update();
+
+    ENGINE.incident().ofInstance(processInstanceKey).withKey(incident.getKey()).resolve();
+
+    // then
+    completeNthJob(processInstanceKey, 2);
+    completeNthJob(processInstanceKey, 3);
+
+    RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_COMPLETED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withElementType(BpmnElementType.PROCESS)
+        .limitToProcessInstanceCompleted()
+        .await();
+  }
+
+  private static void completeNthJob(final long processInstanceKey, final int n) {
+    final var nthJob = findNthJob(processInstanceKey, n);
+    ENGINE.job().withKey(nthJob.getKey()).complete();
+    ENGINE.job().withKey(nthJob.getKey()).complete();
```

Comment:
One job completion should be enough :wink:

Reply:
Haha nice catch!